### PR TITLE
L2 Subdomain Registering

### PR DESCRIPTION
**Pull request by pikonha**
# L2 Subdomain Registering

## All Submissions:

* [X] Have you followed the guidelines in our Contributing document?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

## Description

All the required contracts from ENS to work were deployed to Arbitrum.

image

The L1 resolvers use the EIP-5559 to redirect writing calls to the L2 contracts according to the following diagram:

image

L2 subdomain registering

## Related Issue

#189 

## Changes
- [X] New feature implementation
- [ ] Bug fix
- [ ] Code refactoring
- [ ] Documentation update
- [ ] Other (please specify)

### Changes to Core Features:

* [X] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your core changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

## Additional Notes

The tests will be implemented in a future PR.

### CODE REVIEW

File: packages/contracts/src/NameWrapperProxy.sol
Comment (alextnetto) [inline packages/contracts/src/NameWrapperProxy.sol:1]: I'd change the name of file and contract to `SubdomainController`. I think it brings a better understanding of the role of this contract.

File: packages/contracts/src/ENSIP16.sol [hunk 1]
```diff
@@ -15,26 +15,26 @@ abstract contract ENSIP16 is IERC165, IENSIP16 {
 
     //////// STATE VARIABLES ////////
 
-    string public graphqlUrl;
+    string public metadataUrl;
 
     //////// EVENTS ////////
 
-    event GraphqlUrlSet(string indexed previousUrl, string indexed newUrl);
+    event MetadataUrlSet(string indexed previousUrl, string indexed newUrl);
```
Comment (alextnetto) [inline packages/contracts/src/ENSIP16.sol:22]:
```suggestion
    event MetadataChanged(string indexed previousUrl, string indexed newUrl);
```

File: packages/contracts/src/ArbitrumVerifier.sol [hunk 2]
```diff
@@ -54,17 +54,17 @@ contract ArbVerifier is IEVMVerifier {
...
-        Node memory rblock = rollup.getNode(arbData.nodeIndex);
+        // Node memory rblock = rollup.getNode(arbData.nodeIndex);
 
         //The confirm data is the keccak256 hash of the block hash and the send root. It is used to verify that the rblock is a subject of the layer 2 block that is being proven.
-        bytes32 confirmData = keccak256(
-            abi.encodePacked(
-                keccak256(arbData.rlpEncodedBlock), arbData.sendRoot
-            )
-        );
+        // bytes32 confirmData = keccak256(
```
Comment (alextnetto) [inline packages/contracts/src/ArbitrumVerifier.sol:60]: Create an issue to investigate that!

File: packages/contracts/script/local/L2ArbitrumResolver.s.sol [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+        nameWrapper.registerAndWrapETH2LD(
+            "arb", msg.sender, 31556952000, address(arbResolver), 1
+        );
+
+        bytes[] memory data = new bytes[](1);
+        data[0] = abi.encodeWithSelector(
+            TextResolver.setText.selector,
+            namehash("blockful.arb.eth"),
+            "com.twitter",
+            "@blockfu"
```
Comment (alextnetto) [inline packages/contracts/script/local/L2ArbitrumResolver.s.sol:106]:
```suggestion
            "@blockful"
```

File: packages/client/test/arb.e2e.spec.ts
Comment (alextnetto) [inline packages/client/test/arb.e2e.spec.ts:1]: Open an issue restore these tests and another one to get some metrics on coverage and implement tests good practices.

**Review by alextnetto — COMMENTED** [review]
gj!!

File: packages/contracts/src/ArbitrumVerifier.sol (hunk 2 above)
Comment (pikonha) [reply packages/contracts/src/ArbitrumVerifier.sol:60]: https://github.com/orgs/blockful-io/projects/6?pane=issue&itemId=81962375

File: packages/client/test/arb.e2e.spec.ts
Comment (pikonha) [reply packages/client/test/arb.e2e.spec.ts:1]: https://github.com/orgs/blockful-io/projects/6?pane=issue&itemId=81963074

**Pull request merged**